Mjml.Net/Components/Body/MsoButtonComponent.cs

```diff
@@ -0,0 +1,153 @@
+                    .Attr("role", "presentation")
+                    .Attr("valign", VerticalAlign)
+                    .Style("border", Border)
+                    .Style("border-bottom", BorderBottom)
+                    .Style("border-left", BorderLeft)
+                    .Style("border-right", BorderRight)
+                    .Style("border-top", BorderTop)
+                    .Style("border-radius", BorderRadius)
+                    .Style("cursor", "auto")
+                    .Style("font-style", FontStyle)
+                    .Style("height", Height)
+                    .Style("mso-padding-alt", InnerPadding)
+                    .Style("text-align", TextAlign)
+                    .Style("background", BackgroundColor);
+
+                renderer.StartElement("v:roundrect")
+                    .Attr("xmlns:v", "urn:schemas-microsoft-com:vml")
+                    .Attr("xmlns:w", "urn:schemas-microsoft-com:office:word")
+                    .Attr("fill", hasBackgroundColor ? "t" : "f")
+                    .Attr("strokeweight", stroked ? borderAttributes[0] : "0pt")
+                    .Attr("strokecolor", borderAttributes[2])
+                    .Attr("stroked", stroked ? "t" : "f")
+                    .Attr("arcsize", CalculateArcsize())
+                    .Attr("href", Href)
+                    .Style("height", MsoHeight ?? Height)
+                    .Style("width", MsoWidth ?? Width)
+                    .Style("padding", Padding)
+                    .Style("v-text-anchor", "middle");
+
+                if (stroked)
+                {
+                    renderer.StartElement("v:stroke", true)
+                        .Attr("dashstyle", borderAttributes[1]);
+                }
+
+                if (hasBackgroundColor)
+                {
+                    renderer.StartElement("v:fill", true)
+                        .Attr("color", BackgroundColor);
+                }
+
+                renderer.StartElement("w:anchorlock", true);
+                renderer.StartElement("center");
+
+                RenderButton(renderer);
+
+                renderer.EndElement("center");
+                renderer.EndElement("v:roundrect");
+
+                renderer.EndElement("td");
+                renderer.EndElement("tr");
+            }
+            renderer.Content("<![endif]-->");
+        }
+
+        private string CalculateArcsize()
+        {
+            var radius = UnitParser.Parse(BorderRadius);
+            if (radius.Value <= 0)
+            {
+                return string.Empty;
+            }
+
+            if (radius.Unit == Unit.Percent)
+            {
+                return $"{radius.Value}%";
+            }
+
+            if (radius.Unit == Unit.Pixels)
+            {
+                const string defaultArcsize = "8%";
+                var height = UnitParser.Parse(MsoHeight ?? Height);
+                var percent = Math.Round(radius.Value / height.Value * 100, MidpointRounding.AwayFromZero);
```

Comment:
The percentage should only be calculated if the next condition is true, otherwise it is not needed.

Reply:
refactored into separate method `ConvertBorderRadiusToArcsize`